<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Imputation • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Imputation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/PRONE.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/Normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/Imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Spike_In_Data.html">PRONE with Spike-In Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/daisybio/PRONE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Imputation</h1>
                        <h4 data-toc-skip class="author">Lis Arend</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/daisybio/PRONE/blob/HEAD/vignettes/Imputation.Rmd" class="external-link"><code>vignettes/Imputation.Rmd</code></a></small>
      <div class="d-none name"><code>Imputation.Rmd</code></div>
    </div>

    
    
<div class="section level2" number="1">
<h2 id="load-prone-package">
<span class="header-section-number">1</span> Load PRONE Package<a class="anchor" aria-label="anchor" href="#load-prone-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/daisybio/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="load-data-tmt">
<span class="header-section-number">2</span> Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a> data. For more information on how to create the SummarizedExperiment from a proteomics data set, please refer to the <a href="PRONE.html">“Get Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="preprocessing">
<span class="header-section-number">3</span> Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>As we have seen in the Preprocessing phase, that samples “1.HC_Pool1” and “1.HC_Pool2” have been removed from the data set due to their high amount of missing values (more than 80% of NAs per sample), before imputing the data we will here remove these two samples.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples_manually.html">remove_samples_manually</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Label"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.HC_Pool1"</span>, <span class="st">"1.HC_Pool2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 samples removed.</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 id="missigness-in-proteomics-data">
<span class="header-section-number">4</span> Missigness in Proteomics Data<a class="anchor" aria-label="anchor" href="#missigness-in-proteomics-data"></a>
</h2>
<p>Since proteomics data is often affected by missing values and some statistical tests do not allow a high amount of missingness in the data, people have to options to reduce the amount of missingness in the data: (1) remove proteins with missing values or (2) impute missing values.</p>
<p>(1): this point is already shown in the Preprocessing tutorial, where we removed samples with a high amount of missing values using a predefined threshold.</p>
<p>(2): this point will be discussed here.</p>
</div>
<div class="section level2" number="5">
<h2 id="impute-data">
<span class="header-section-number">5</span> Impute Data<a class="anchor" aria-label="anchor" href="#impute-data"></a>
</h2>
<p>Since the initial focus of PRONE was on the evaluation of the performance of normalization methods and a selection of methods was made based on an extensive literature review, the imputation methods are currently still limited. However, to ensure that PRONE offers all steps of a typical proteomics analysis workflow, we have included a basic imputation method since in some cases imputation is favored over removing a high amount of proteins.</p>
<p>So currently, there is only a mixed imputation method available in PRONE: k-nearest neighbor imputation for proteins with missing values at random and a left-shifted Gaussian distribution for proteins with missing values not at random. Imputation can be performed on a selection of normalized data sets using the “ain” parameter in the <code>impute_SE</code> function. The default is to impute all assays (ain = NULL).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute_se.html">impute_se</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span></code></pre></div>
<p>ATTENTION:</p>
<p>Please note that imputation can introduce bias in the data and should be used with caution. After imputing your data, have a look at the exploratory data analysis plots (such as boxplots, PCA plots, etc.) to see if the imputation method has skewed the distributions of your samples and introduced biases in your data. These visualizations options are already shown in the <a href="Normalization.html">“Normalization”</a> tutorial.</p>
</div>
<div class="section level2" number="6">
<h2 id="session-info">
<span class="header-section-number">6</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] PRONE_1.0.4                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.64.0              GenomicRanges_1.56.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.40.1         IRanges_2.38.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.42.1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [3] clue_0.3-65                 compiler_4.4.1             </span></span>
<span><span class="co">#&gt;  [5] vctrs_0.6.5                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [7] stringr_1.5.1               ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt;  [9] pkgconfig_2.0.3             crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [11] fastmap_1.2.0               XVector_0.44.0             </span></span>
<span><span class="co">#&gt; [13] utf8_1.2.4                  rmarkdown_2.27             </span></span>
<span><span class="co">#&gt; [15] UCSC.utils_1.0.0            preprocessCore_1.66.0      </span></span>
<span><span class="co">#&gt; [17] purrr_1.0.2                 xfun_0.46                  </span></span>
<span><span class="co">#&gt; [19] MultiAssayExperiment_1.30.3 zlibbioc_1.50.0            </span></span>
<span><span class="co">#&gt; [21] cachem_1.1.0                jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [23] DelayedArray_0.30.1         BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt; [25] parallel_4.4.1              cluster_2.1.6              </span></span>
<span><span class="co">#&gt; [27] R6_2.5.1                    bslib_0.8.0                </span></span>
<span><span class="co">#&gt; [29] stringi_1.8.4               limma_3.60.4               </span></span>
<span><span class="co">#&gt; [31] jquerylib_0.1.4             Rcpp_1.0.13                </span></span>
<span><span class="co">#&gt; [33] bookdown_0.40               iterators_1.0.14           </span></span>
<span><span class="co">#&gt; [35] knitr_1.48                  Matrix_1.7-0               </span></span>
<span><span class="co">#&gt; [37] igraph_2.0.3                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt; [39] rstudioapi_0.16.0           abind_1.4-5                </span></span>
<span><span class="co">#&gt; [41] yaml_2.3.10                 doParallel_1.0.17          </span></span>
<span><span class="co">#&gt; [43] ggtext_0.1.2                codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [45] affy_1.82.0                 lattice_0.22-6             </span></span>
<span><span class="co">#&gt; [47] tibble_3.2.1                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [49] withr_3.0.1                 evaluate_0.24.0            </span></span>
<span><span class="co">#&gt; [51] xml2_1.3.6                  pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [53] affyio_1.74.0               BiocManager_1.30.23        </span></span>
<span><span class="co">#&gt; [55] renv_1.0.7                  foreach_1.5.2              </span></span>
<span><span class="co">#&gt; [57] MSnbase_2.30.1              MALDIquant_1.22.2          </span></span>
<span><span class="co">#&gt; [59] ncdf4_1.22                  generics_0.1.3             </span></span>
<span><span class="co">#&gt; [61] ggplot2_3.5.1               munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [63] scales_1.3.0                glue_1.7.0                 </span></span>
<span><span class="co">#&gt; [65] lazyeval_0.2.2              tools_4.4.1                </span></span>
<span><span class="co">#&gt; [67] data.table_1.15.4           mzID_1.42.0                </span></span>
<span><span class="co">#&gt; [69] QFeatures_1.14.2            vsn_3.72.0                 </span></span>
<span><span class="co">#&gt; [71] mzR_2.38.0                  XML_3.99-0.17              </span></span>
<span><span class="co">#&gt; [73] grid_4.4.1                  impute_1.78.0              </span></span>
<span><span class="co">#&gt; [75] tidyr_1.3.1                 MsCoreUtils_1.16.1         </span></span>
<span><span class="co">#&gt; [77] colorspace_2.1-1            GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt; [79] PSMatch_1.8.0               cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [81] fansi_1.0.6                 S4Arrays_1.4.1             </span></span>
<span><span class="co">#&gt; [83] dplyr_1.1.4                 AnnotationFilter_1.28.0    </span></span>
<span><span class="co">#&gt; [85] pcaMethods_1.96.0           gtable_0.3.5               </span></span>
<span><span class="co">#&gt; [87] sass_0.4.9                  digest_0.6.36              </span></span>
<span><span class="co">#&gt; [89] SparseArray_1.4.8           htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [91] htmltools_0.5.8.1           lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [93] httr_1.4.7                  statmod_1.5.0              </span></span>
<span><span class="co">#&gt; [95] gridtext_0.1.5              MASS_7.3-61</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes in Infected Patients Depending on Tuberculosis Disease State.”</span> <em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
