<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Normalization • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Normalization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.16</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/PRONE.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/Normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/Imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Spike_In_Data.html">PRONE with Spike-In Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Normalization</h1>
                        <h4 data-toc-skip class="author">Lis Arend</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Normalization.Rmd" class="external-link"><code>vignettes/Normalization.Rmd</code></a></small>
      <div class="d-none name"><code>Normalization.Rmd</code></div>
    </div>

    
    
<div class="section level2" number="1">
<h2 id="load-prone-package">
<span class="header-section-number">1</span> Load PRONE Package<a class="anchor" aria-label="anchor" href="#load-prone-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and attach PRONE</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="load-data-tmt">
<span class="header-section-number">2</span> Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a> data. For more information on how to create the SummarizedExperiment from a proteomics data set, please refer to the <a href="PRONE.html">“Get Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
<p>This SummarizedExperiment object already includes data of different normalization methods. Since this vignette should show you how to use the PRONE workflow for novel proteomics data, we will remove the normalized data and only keep the raw and log2 data that are available after loading the data accordingly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"raw"</span>, <span class="st">"log2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="normalization">
<span class="header-section-number">3</span> Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<p>For normalization, there are multiple functions available which can be used to normalize the data. First of all, to know which normalization methods can be applied:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_normalization_methods.html">get_normalization_methods</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "GlobalMean"    "GlobalMedian"  "Median"        "Mean"         </span></span>
<span><span class="co">#&gt;  [5] "IRS"           "Quantile"      "VSN"           "LoessF"       </span></span>
<span><span class="co">#&gt;  [9] "LoessCyc"      "RLR"           "RlrMA"         "RlrMACyc"     </span></span>
<span><span class="co">#&gt; [13] "EigenMS"       "MAD"           "RobNorm"       "TMM"          </span></span>
<span><span class="co">#&gt; [17] "limBE"         "NormicsVSN"    "NormicsMedian"</span></span></code></pre></div>
<p>You can use <code><a href="../reference/normalize_se.html">normalize_se()</a></code> by specifying all normalization methods you want to apply on the data. For instance, if you want to perform RobNorm, median, and mean normalization, just execute this line:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RobNorm"</span>, <span class="st">"Mean"</span>, <span class="st">"Median"</span><span class="op">)</span>, </span>
<span>                        combination_pattern <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; RobNorm completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; Median completed.</span></span></code></pre></div>
<p>Tandem mass tag (TMT), a chemical labeling method that enables the simultaneous MS-analysis of up to 18 samples pooled together, is increasingly being applied in large-scale proteomic studies. Since the integration of multiple TMT batches within a single analysis leads to high batch variation and affects data quality, a batch effect correction method, such as internal reference scaling (IRS) or the <code>limma::removeBatchEffects</code> method (in PRONE: limBE), is required to adjust for these systematic biases. Commonly, batch effect correction is applied after basic normalization. However, the order of normalization and batch effect correction can be changed in PRONE.</p>
<p>For instance, if you want to perform IRS to reduce the batch effects on top of the previously normalized data, you can use the combination pattern “<em>on</em>”.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RobNorm"</span>, <span class="st">"Mean"</span>, <span class="st">"Median"</span>, <span class="st">"IRS_on_RobNorm"</span>, </span>
<span>                        <span class="st">"IRS_on_Mean"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>                        combination_pattern <span class="op">=</span> <span class="st">"_on_"</span><span class="op">)</span></span>
<span><span class="co">#&gt; RobNorm completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on RobNorm-normalized data completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on Mean-normalized data completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on Median-normalized data completed.</span></span></code></pre></div>
<p>Finally, you can also normalize your data by applying the specific normalization method. This makes it possible to design the parameters of an individual function more specifically. For instance, if you want to normalize the data by the median, you can use the function <code><a href="../reference/medianNorm.html">medianNorm()</a></code>. By default, median normalization is performed on raw-data. Using the individual normalization functions, you can easily specify</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/medianNorm.html">medianNorm</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, aout <span class="op">=</span> <span class="st">"Median"</span><span class="op">)</span></span></code></pre></div>
<p>All normalized intensities are stored in the SummarizedExperiment object and you can check the already performed normalization techniques using:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se_norm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "raw"            "log2"           "RobNorm"        "Mean"          </span></span>
<span><span class="co">#&gt; [5] "Median"         "IRS_on_RobNorm" "IRS_on_Mean"    "IRS_on_Median"</span></span></code></pre></div>
<p>We suggest using the default value of the on_raw parameter. This parameter specifies whether the data should be normalized on raw or log2-transformed data. The default value of the “on_raw” parameters was made for each normalization method individually based on publications.</p>
</div>
<div class="section level2" number="4">
<h2 id="qualitative-and-quantitative-evaluation">
<span class="header-section-number">4</span> Qualitative and Quantitative Evaluation<a class="anchor" aria-label="anchor" href="#qualitative-and-quantitative-evaluation"></a>
</h2>
<p>Sample distributions are often skewed due to the systematic biases introduced throughout all steps of an MS-experiment. To evaluate the performance of the normalization methods, the distribution of the normalized data can be visualized using boxplots. One would expect to align the sample distributions more closely after normalization.</p>
<p>Moreover, since normalization tries to remove the technical bias while keeping the biological variation, PCA plots of the log2-transformed data should be used to analyze if the technical bias is more prominent than the biological variation. After normalization, the samples should cluster according to the biological groups rather than the technical biases. So it may be helpful to analyze the PCA plots of the different normalization techniques to see which normalization method mostly reduced the technical bias.</p>
<p>Finally, the assessment of the normalization methods is commonly centered on their ability to decrease intragroup variation between samples, using intragroup pooled coefficient of variation (PCV), pooled estimate of variance (PEV), and pooled median absolute deviation (PMAD) as measures of variability. Furthermore, the Pearson correlation between samples is used to measure the similarity of the samples within the same group. A normalization method should reduce intragroup variation between samples and increases the correlation between samples within the same group. However, please do not focus solely on intragroup variation, but also consider the other evaluation methods and perform differential expression analysis to further evaluate the methods and analyze the impact of normalization on the downstream results. More details on how to evaluate the performance of the normalization techniques are coming along with our paper. Once it is published, it will be referenced here.</p>
<p>Hence, PRONE offers many functions to comparatively evaluate the performance of the normalization methods. Notably, the parameter “ain” can always be set. By specifying “ain = NULL”, all normalization methods that were previously performed and are saved in the SummarizedExperiment object are considered. If you want to evaluate only a selection of normalization methods, you can specify them in the “ain” parameter. For instance, if you want to evaluate only the IRS_on_RobNorm and Mean normalization methods, you can set “ain = c(”IRS_on_RobNorm”, “Mean”)“.</p>
<div class="section level3" number="4.1">
<h3 id="visual-inspection">
<span class="header-section-number">4.1</span> Visual Inspection<a class="anchor" aria-label="anchor" href="#visual-inspection"></a>
</h3>
<p>You can comparatively visualize the normalized data by using the function <code><a href="../reference/plot_boxplots.html">plot_boxplots()</a></code>, <code><a href="../reference/plot_densities.html">plot_densities()</a></code>, and <code>plot_pca()</code>.</p>
<div class="section level4" number="4.1.1">
<h4 id="boxplots-of-normalized-data">
<span class="header-section-number">4.1.1</span> Boxplots of Normalized Data<a class="anchor" aria-label="anchor" href="#boxplots-of-normalized-data"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>              label_by <span class="op">=</span> <span class="cn">NULL</span>, ncol <span class="op">=</span> <span class="fl">3</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>, legend.direction <span class="op">=</span> <span class="st">"horizontal"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-9-1.png" alt="Boxplots of protein intensities of all normalization methods of the SummarizedExperiment object. This plot shows the protein distributions across samples." width="768"><p class="caption">
Figure 4.1: Boxplots of protein intensities of all normalization methods of the SummarizedExperiment object. This plot shows the protein distributions across samples.
</p>
</div>
<p>But you can also just plot a selection of normalization methods and color for instance by batch:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>              color_by <span class="op">=</span> <span class="st">"Pool"</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-10-1.png" alt="Boxplots of protein intensities of a selection of normalization methods and colored by batch. This plot shows the protein distributions across samples." width="768"><p class="caption">
Figure 4.2: Boxplots of protein intensities of a selection of normalization methods and colored by batch. This plot shows the protein distributions across samples.
</p>
</div>
<p>Another option that you have is to return the boxplots for each normalized data independently as a single ggplot object. For this, you need to set facet = FALSE:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>              color_by <span class="op">=</span> <span class="st">"Pool"</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $IRS_on_RobNorm</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-11-1"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-11-1.png" alt="Single boxplot of protein intensities per selected normalization method and colored by batch .This plot shows the protein distributions across samples." width="768"><p class="caption">
Figure 4.3: Single boxplot of protein intensities per selected normalization method and colored by batch .This plot shows the protein distributions across samples.
</p>
</div>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $IRS_on_Median</span></span></code></pre>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-11-2"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-11-2.png" alt="Single boxplot of protein intensities per selected normalization method and colored by batch .This plot shows the protein distributions across samples." width="768"><p class="caption">
Figure 4.4: Single boxplot of protein intensities per selected normalization method and colored by batch .This plot shows the protein distributions across samples.
</p>
</div>
</div>
<div class="section level4" number="4.1.2">
<h4 id="densities-of-normalized-data">
<span class="header-section-number">4.1.2</span> Densities of Normalized Data<a class="anchor" aria-label="anchor" href="#densities-of-normalized-data"></a>
</h4>
<p>Similarly you can visualize the densities of the normalized data.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_densities.html">plot_densities</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>               color_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-12-1.png" alt="Density plot of protein intensities of for a selection of normalization method." width="768"><p class="caption">
Figure 4.5: Density plot of protein intensities of for a selection of normalization method.
</p>
</div>
</div>
<div class="section level4" number="4.1.3">
<h4 id="pca-of-normalized-data">
<span class="header-section-number">4.1.3</span> PCA of Normalized Data<a class="anchor" aria-label="anchor" href="#pca-of-normalized-data"></a>
</h4>
<p>Furthermore, you can visualize the normalized data in a PCA plot. Here you have some more arguments that can be specified If you decide to visualize the methods in independent plots (facet_norm = FALSE), then a list of ggplot objects is returned. However, you have the additional option to facet by any other column of the metadata (using the facet_by parameter). Here an example:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>         color_by <span class="op">=</span> <span class="st">"Group"</span>, label_by <span class="op">=</span> <span class="st">"No"</span>, shape_by <span class="op">=</span> <span class="st">"Pool"</span>, </span>
<span>         facet_norm <span class="op">=</span> <span class="cn">FALSE</span>, facet_by <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span><span class="co">#&gt; No labeling of samples.</span></span>
<span><span class="co">#&gt; $IRS_on_RobNorm</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-13-1"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-13-1.png" alt="Single PCA plot per normalization method, colored and faceted by condition, and shaped by batch effect. This plot helps to identify batch effects." width="768"><p class="caption">
Figure 4.6: Single PCA plot per normalization method, colored and faceted by condition, and shaped by batch effect. This plot helps to identify batch effects.
</p>
</div>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $IRS_on_Median</span></span></code></pre>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-13-2"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-13-2.png" alt="Single PCA plot per normalization method, colored and faceted by condition, and shaped by batch effect. This plot helps to identify batch effects." width="768"><p class="caption">
Figure 4.7: Single PCA plot per normalization method, colored and faceted by condition, and shaped by batch effect. This plot helps to identify batch effects.
</p>
</div>
<p>Or you can simply plot the PCA of the normalized data next to each other. However, the facet_by argument can then not be used. Reminder, by setting color_by = NULL, it will be first checked if a condition has been set in the SummarizedExperiment during loading the data.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span>, </span>
<span>         color_by <span class="op">=</span> <span class="st">"Group"</span>, label_by <span class="op">=</span> <span class="st">"No"</span>, shape_by <span class="op">=</span> <span class="st">"Pool"</span>, </span>
<span>         facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; No labeling of samples.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-14-1.png" alt="Combined PCA plot of a selection of normalization methods, colored by condition and shaped by batch effect. This plot helps to identify batch effects." width="768"><p class="caption">
Figure 4.8: Combined PCA plot of a selection of normalization methods, colored by condition and shaped by batch effect. This plot helps to identify batch effects.
</p>
</div>
</div>
</div>
<div class="section level3" number="4.2">
<h3 id="intragroup-variation">
<span class="header-section-number">4.2</span> Intragroup Variation<a class="anchor" aria-label="anchor" href="#intragroup-variation"></a>
</h3>
<p>In PRONE, you can evaluate the intragroup variation of the normalized data by using the functions <code><a href="../reference/plot_intragroup_correlation.html">plot_intragroup_correlation()</a></code>, <code><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV()</a></code>, <code><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD()</a></code>, and <code><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_correlation.html">plot_intragroup_correlation</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                            method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-15"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-15-1.png" alt="Intragroup similarity measured by Pearson correlation for all normalization methods of the SummarizedExperiment object. The correlation between samples within the same group was calculated for all normalization methods and sample groups.." width="768"><p class="caption">
Figure 4.9: Intragroup similarity measured by Pearson correlation for all normalization methods of the SummarizedExperiment object. The correlation between samples within the same group was calculated for all normalization methods and sample groups..
</p>
</div>
<p>You have two options to visualize intragroup PCV, PEV, and PMAD. You can either simply generate boxplots of intragroup variation of each normalization method (diff = FALSE), or you can visualize the reduction of intragroup variation of each normalization method compared to log2 (diff = TRUE).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-16"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-16-1.png" alt="Intragroup variation by calculation of the pooled coefficient of variation (PCV) for all normalization methods of the SummarizedExperiment object. The PCV was calculated for all normalization methods and sample groups." width="768"><p class="caption">
Figure 4.10: Intragroup variation by calculation of the pooled coefficient of variation (PCV) for all normalization methods of the SummarizedExperiment object. The PCV was calculated for all normalization methods and sample groups.
</p>
</div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-17"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-17-1.png" alt="Intragroup variation by calculation of the pooled estimate of variance (PEV) for all normalization methods of the SummarizedExperiment object. The PEV was calculated for all normalization methods and sample groups." width="768"><p class="caption">
Figure 4.11: Intragroup variation by calculation of the pooled estimate of variance (PEV) for all normalization methods of the SummarizedExperiment object. The PEV was calculated for all normalization methods and sample groups.
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-18-1.png" alt="Intragroup variation by calculation of the pooled median absolute deviation (PMAD) for all normalization methods of the SummarizedExperiment object. The PMAD was calculated for all normalization methods and sample groups." width="768"><p class="caption">
Figure 4.12: Intragroup variation by calculation of the pooled median absolute deviation (PMAD) for all normalization methods of the SummarizedExperiment object. The PMAD was calculated for all normalization methods and sample groups.
</p>
</div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-19-1.png" alt="Reduction of intragroup variation, here using PCV, for all normalization methods of the SummarizedExperiment object compared to log2." width="768"><p class="caption">
Figure 4.13: Reduction of intragroup variation, here using PCV, for all normalization methods of the SummarizedExperiment object compared to log2.
</p>
</div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-20"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-20-1.png" alt="Reduction of intragroup variation, here using PEV, for all normalization methods of the SummarizedExperiment object compared to log2." width="768"><p class="caption">
Figure 4.14: Reduction of intragroup variation, here using PEV, for all normalization methods of the SummarizedExperiment object compared to log2.
</p>
</div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-21"></span>
<img src="Normalization_files/figure-html/unnamed-chunk-21-1.png" alt="Reduction of intragroup variation, here using PMAD, for all normalization methods of the SummarizedExperiment object compared to log2." width="768"><p class="caption">
Figure 4.15: Reduction of intragroup variation, here using PMAD, for all normalization methods of the SummarizedExperiment object compared to log2.
</p>
</div>
</div>
</div>
<div class="section level2" number="5">
<h2 id="subset-summarizedexperiment">
<span class="header-section-number">5</span> Subset SummarizedExperiment<a class="anchor" aria-label="anchor" href="#subset-summarizedexperiment"></a>
</h2>
<p>After the qualitative and quantitative evaluation, you may have noticed that some normalization techniques are not appropriate for the specific real-world data set. For further analysis, you want to remove them and not evaluate the specific normalization methods furthermore. For this, the <code><a href="../reference/remove_assays_from_SE.html">remove_assays_from_SE()</a></code> method can be used.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_no_MAD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_assays_from_SE.html">remove_assays_from_SE</a></span><span class="op">(</span><span class="va">se_norm</span>, assays_to_remove <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Mean"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>In contrast, you can also subset the SummarizedExperiment object to only include specific normalization techniques using the <code><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm()</a></code>method.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm</a></span><span class="op">(</span><span class="va">se_norm</span>, </span>
<span>                               ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="6">
<h2 id="session-info">
<span class="header-section-number">6</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] PRONE_0.99.16               SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.64.0              GenomicRanges_1.56.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.40.1         IRanges_2.38.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.42.1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] clue_0.3-65                 compiler_4.4.1             </span></span>
<span><span class="co">#&gt;   [5] vctrs_0.6.5                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;   [7] stringr_1.5.1               ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt;   [9] pkgconfig_2.0.3             crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [11] fastmap_1.2.0               XVector_0.44.0             </span></span>
<span><span class="co">#&gt;  [13] labeling_0.4.3              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.27              UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt;  [17] preprocessCore_1.66.0       purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [19] xfun_0.46                   MultiAssayExperiment_1.30.3</span></span>
<span><span class="co">#&gt;  [21] zlibbioc_1.50.0             cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [23] jsonlite_1.8.8              highr_0.11                 </span></span>
<span><span class="co">#&gt;  [25] DelayedArray_0.30.1         BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt;  [27] parallel_4.4.1              cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [29] R6_2.5.1                    RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [31] bslib_0.8.0                 stringi_1.8.4              </span></span>
<span><span class="co">#&gt;  [33] limma_3.60.4                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [35] Rcpp_1.0.13                 bookdown_0.40              </span></span>
<span><span class="co">#&gt;  [37] iterators_1.0.14            knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [39] Matrix_1.7-0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt;  [41] tidyselect_1.2.1            rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [43] abind_1.4-5                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [45] doParallel_1.0.17           ggtext_0.1.2               </span></span>
<span><span class="co">#&gt;  [47] codetools_0.2-20            affy_1.82.0                </span></span>
<span><span class="co">#&gt;  [49] lattice_0.22-6              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [51] plyr_1.8.9                  withr_3.0.1                </span></span>
<span><span class="co">#&gt;  [53] evaluate_0.24.0             xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [55] pillar_1.9.0                affyio_1.74.0              </span></span>
<span><span class="co">#&gt;  [57] BiocManager_1.30.23         renv_1.0.7                 </span></span>
<span><span class="co">#&gt;  [59] foreach_1.5.2               MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt;  [61] MALDIquant_1.22.2           ncdf4_1.22                 </span></span>
<span><span class="co">#&gt;  [63] generics_0.1.3              ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [65] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [67] glue_1.7.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [69] tools_4.4.1                 data.table_1.15.4          </span></span>
<span><span class="co">#&gt;  [71] mzID_1.42.0                 QFeatures_1.14.2           </span></span>
<span><span class="co">#&gt;  [73] vsn_3.72.0                  mzR_2.38.0                 </span></span>
<span><span class="co">#&gt;  [75] XML_3.99-0.17               grid_4.4.1                 </span></span>
<span><span class="co">#&gt;  [77] impute_1.78.0               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [79] MsCoreUtils_1.16.1          colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [81] GenomeInfoDbData_1.2.12     PSMatch_1.8.0              </span></span>
<span><span class="co">#&gt;  [83] cli_3.6.3                   fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [85] S4Arrays_1.4.1              dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [87] AnnotationFilter_1.28.0     pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt;  [89] gtable_0.3.5                sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [91] digest_0.6.36               SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt;  [93] farver_2.1.2                htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [95] htmltools_0.5.8.1           lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [97] httr_1.4.7                  NormalyzerDE_1.22.0        </span></span>
<span><span class="co">#&gt;  [99] statmod_1.5.0               gridtext_0.1.5             </span></span>
<span><span class="co">#&gt; [101] MASS_7.3-61</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes in Infected Patients Depending on Tuberculosis Disease State.”</span> <em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
