<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Preprocessing • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Preprocessing">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/PRONE.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/Normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/Imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Spike_In_Data.html">PRONE with Spike-In Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Preprocessing</h1>
                        <h4 data-toc-skip class="author">Lis Arend</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Preprocessing.Rmd" class="external-link"><code>vignettes/Preprocessing.Rmd</code></a></small>
      <div class="d-none name"><code>Preprocessing.Rmd</code></div>
    </div>

    
    
<div class="section level2" number="1">
<h2 id="load-prone-package">
<span class="header-section-number">1</span> Load PRONE Package<a class="anchor" aria-label="anchor" href="#load-prone-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and attach PRONE</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="load-data-tmt">
<span class="header-section-number">2</span> Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a> data. For more information on how to create the SummarizedExperiment from a proteomics data set, please refer to the <a href="PRONE.html">“Get Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
<p>This SummarizedExperiment object already includes data of different normalization methods. Since this vignette should show you how to use the PRONE workflow for novel proteomics data, we will remove the normalized data and only keep the raw and log2 data that are available after loading the data accordingly.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"raw"</span>, <span class="st">"log2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="overview-of-the-data">
<span class="header-section-number">3</span> Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h2>
<p>To get an overview on the number of NAs, you can simply use the function <code><a href="../reference/get_NA_overview.html">get_NA_overview()</a></code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_NA_overview.html">get_NA_overview</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">Total.Values</th>
<th align="right">NA.Values</th>
<th align="right">NA.Percentage</th>
</tr></thead>
<tbody><tr class="odd">
<td align="right">6020</td>
<td align="right">1945</td>
<td align="right">32.30897</td>
</tr></tbody>
</table>
<p>To get an overview on the number of samples per sample group or batch, you can simply use the function <code><a href="../reference/plot_condition_overview.html">plot_condition_overview()</a></code> by specifying the column of the meta-data that should be used for coloring. By default (condition = NULL), the column specified in <code><a href="../reference/load_data.html">load_data()</a></code> will be used.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-4-1.png" alt="Overview barplot of the number of samples per condition." width="768"><p class="caption">
Figure 3.1: Overview barplot of the number of samples per condition.
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="st">"Pool"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-5-1.png" alt="Overview barplot of the number of samples per pool." width="768"><p class="caption">
Figure 3.2: Overview barplot of the number of samples per pool.
</p>
</div>
<p>A general overview of the protein intensities across the different samples is provided by the function <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>. The parameter “ain” specifies the data to plot, currently only “raw” and “log2” is available (names(assays(se)). Later if multiple normalization methods are executed, these will be saved as assays, and the normalized data can be visualized.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">available_ains</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, color_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pool"</span>, <span class="st">"Group"</span><span class="op">)</span>, </span>
<span>             label_by <span class="op">=</span> <span class="cn">NULL</span>, only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $log2</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-6-1.png" alt="Heatmap of the log2-protein intensities with columns and proteins being clustered with hclust." width="768"><p class="caption">
Figure 3.3: Heatmap of the log2-protein intensities with columns and proteins being clustered with hclust.
</p>
</div>
<p>Similarly, an upset plot can be generated to visualize the overlaps between sets defined by a specific column in the metadata. The sets are generated by using non-NA values.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, mb.ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>,<span class="fl">0.3</span><span class="op">)</span>, </span>
<span>           only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-7-1.png" alt="Upset plot of the non-NA protein intensities with sets defined by the Pool column." width="768"><p class="caption">
Figure 3.4: Upset plot of the non-NA protein intensities with sets defined by the Pool column.
</p>
</div>
<p>If you are interested in the intensities of specific biomarkers, you can use the <code><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots()</a></code> function to compare the distribution of intensities per group. The plot can be generated per marker and facet by normalization method (facet_norm = TRUE) or by normalization method and facet by marker (facet_marker = TRUE).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots</a></span><span class="op">(</span><span class="va">se</span>, </span>
<span>                           markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q92954;J3KP74;E9PLR3"</span>, <span class="st">"Q9Y6Z7"</span><span class="op">)</span>, </span>
<span>                           ain <span class="op">=</span> <span class="st">"log2"</span>, </span>
<span>                           id_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, </span>
<span>                           facet_norm <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                           facet_marker <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; No shaping done.</span></span>
<span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-8-1.png" alt="Boxplot of the log2-protein intensities of the markers Q92954;J3KP74;E9PLR3, Q9Y6Z7, and Q68CQ4. If specific biomarkers are known to be upregulated in some conditions, you can use this plot to get an impression of the intensities before doing differential expression analysis." width="768"><p class="caption">
Figure 3.5: Boxplot of the log2-protein intensities of the markers Q92954;J3KP74;E9PLR3, Q9Y6Z7, and Q68CQ4. If specific biomarkers are known to be upregulated in some conditions, you can use this plot to get an impression of the intensities before doing differential expression analysis.
</p>
</div>
</div>
<div class="section level2" number="4">
<h2 id="filter-proteins">
<span class="header-section-number">4</span> Filter Proteins<a class="anchor" aria-label="anchor" href="#filter-proteins"></a>
</h2>
<div class="section level3" number="4.1">
<h3 id="remove-proteins-with-missing-values-in-all-samples">
<span class="header-section-number">4.1</span> Remove Proteins With Missing Values in ALL Samples<a class="anchor" aria-label="anchor" href="#remove-proteins-with-missing-values-in-all-samples"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_complete_NA_proteins.html">filter_out_complete_NA_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 13 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level3" number="4.2">
<h3 id="remove-proteins-with-a-specific-value-in-a-specific-column">
<span class="header-section-number">4.2</span> Remove Proteins With a Specific Value in a Specific Column<a class="anchor" aria-label="anchor" href="#remove-proteins-with-a-specific-value-in-a-specific-column"></a>
</h3>
<p>Typically proteins with “+” in the columns “Reverse”, “Only.identified.by.site”, and “Potential.contaminant” are removed in case of a MaxQuant proteinGroups.txt output file.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_value.html">filter_out_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Reverse"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 17 proteins were removed.</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_value.html">filter_out_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Only.identified.by.site"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 proteins were removed.</span></span>
<span><span class="co">#se &lt;- filter_out_proteins_by_value(se, "Potential.contaminant", "+")</span></span></code></pre></div>
</div>
<div class="section level3" number="4.3">
<h3 id="remove-proteins-by-id">
<span class="header-section-number">4.3</span> Remove Proteins by ID<a class="anchor" aria-label="anchor" href="#remove-proteins-by-id"></a>
</h3>
<p>If you don’t want to remove for instance all proteins with “Potential.contaminant == +”, you can also first get the protein ID with the specific value, check them in Uniprot, and then remove only some by using the function <code><a href="../reference/filter_out_proteins_by_ID.html">filter_out_proteins_by_ID()</a></code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pot_contaminants</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_proteins_by_value.html">get_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Potential.contaminant"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were identified.</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_ID.html">filter_out_proteins_by_ID</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">pot_contaminants</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level3" number="4.4">
<h3 id="explore-missing-value-pattern">
<span class="header-section-number">4.4</span> Explore Missing Value Pattern<a class="anchor" aria-label="anchor" href="#explore-missing-value-pattern"></a>
</h3>
<p>Due to the high amount of missing values in MS-based proteomics data, it is important to explore the missing value pattern in the data. The function <code><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap()</a></code> provides a heatmap of the proteins with at least one missing value across all samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, cluster_samples <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                cluster_proteins <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-12-1.png" alt="Heatmap of the missing value pattern. Proteins with at least one missing value are included in this heatmap." width="768"><p class="caption">
Figure 4.1: Heatmap of the missing value pattern. Proteins with at least one missing value are included in this heatmap.
</p>
</div>
<p>Another way to explore the missing value pattern is to use the functions <code><a href="../reference/plot_NA_density.html">plot_NA_density()</a></code> and <code><a href="../reference/plot_NA_frequency.html">plot_NA_frequency()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_density.html">plot_NA_density</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-13"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-13-1.png" alt="Density plot of protein intensities of proteins with missing values and proteins without missing values. This can help to check whether the missing values are biased to lower intense proteins. In this case, the missing values are biased towards the lower intense proteins." width="768"><p class="caption">
Figure 4.2: Density plot of protein intensities of proteins with missing values and proteins without missing values. This can help to check whether the missing values are biased to lower intense proteins. In this case, the missing values are biased towards the lower intense proteins.
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_frequency.html">plot_NA_frequency</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-14-1.png" alt="Protein identification overlap. This plot shows the number of identified proteins (y-axis) per number of samples (x-axis)." width="768"><p class="caption">
Figure 4.3: Protein identification overlap. This plot shows the number of identified proteins (y-axis) per number of samples (x-axis).
</p>
</div>
</div>
<div class="section level3" number="4.5">
<h3 id="filter-proteins-by-applying-a-missing-value-threshold">
<span class="header-section-number">4.5</span> Filter Proteins By Applying a Missing Value Threshold<a class="anchor" aria-label="anchor" href="#filter-proteins-by-applying-a-missing-value-threshold"></a>
</h3>
<p>To reduce the amount of missing values, it is possible to filter proteins by applying a missing value threshold. The function <code><a href="../reference/filter_out_NA_proteins_by_threshold.html">filter_out_NA_proteins_by_threshold()</a></code> removes proteins with more missing values than the specified threshold. The threshold is a value between 0 and 1, where 0.7, for instance, means that proteins with less than 70% of real values will be removed, i.e., proteins with more than 30% missing values will be removed.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_NA_proteins_by_threshold.html">filter_out_NA_proteins_by_threshold</a></span><span class="op">(</span><span class="va">se</span>, thr <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> </span>
<span><span class="co">#&gt; 99 proteins were removed.</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-15"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-15-1.png" alt="Heatmap of the missing value pattern after filtering proteins with too many missing values (threshold was set to 70%). Proteins with at least one missing value are included in this heatmap." width="768"><p class="caption">
Figure 4.4: Heatmap of the missing value pattern after filtering proteins with too many missing values (threshold was set to 70%). Proteins with at least one missing value are included in this heatmap.
</p>
</div>
</div>
</div>
<div class="section level2" number="5">
<h2 id="filter-samples">
<span class="header-section-number">5</span> Filter Samples<a class="anchor" aria-label="anchor" href="#filter-samples"></a>
</h2>
<p>Following filtering proteins by different criteria, samples can be analyzed more in detail. PRONE provides some functions, such as <code><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples()</a></code> and <code><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples()</a></code>, to get an overview of the number of proteins and the total intensity per sample, but also offers the automatic outlier detection method of POMA.</p>
<div class="section level3" number="5.1">
<h3 id="quality-control">
<span class="header-section-number">5.1</span> Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-16"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-16-1.png" alt="Barplot on the number of non-zero proteins per samples colored by condition. This plot helps as quality control to check whether the number of proteins is similar across samples. In this case, only very few proteins have been detected for sample 1.HC_Pool1 compared to the other samples." width="768"><p class="caption">
Figure 5.1: Barplot on the number of non-zero proteins per samples colored by condition. This plot helps as quality control to check whether the number of proteins is similar across samples. In this case, only very few proteins have been detected for sample 1.HC_Pool1 compared to the other samples.
</p>
</div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-17"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-17-1.png" alt="Barplot on the total protein intensity per samples colored by condition. This plot helps as quality control to check whether the total protein intensity is similar across samples. In this case, the total protein intensities of samples 1.HC_Pool1 and 1.HC_Pool2 are much lower compared to the other samples." width="768"><p class="caption">
Figure 5.2: Barplot on the total protein intensity per samples colored by condition. This plot helps as quality control to check whether the total protein intensity is similar across samples. In this case, the total protein intensities of samples 1.HC_Pool1 and 1.HC_Pool2 are much lower compared to the other samples.
</p>
</div>
</div>
<div class="section level3" number="5.2">
<h3 id="remove-samples-manually">
<span class="header-section-number">5.2</span> Remove Samples Manually<a class="anchor" aria-label="anchor" href="#remove-samples-manually"></a>
</h3>
<p>Based on these plots, samples “1.HC_Pool1” and 1_HC_Pool2 seem to be outliers. You can easily remove samples manually by using the <code><a href="../reference/remove_samples_manually.html">remove_samples_manually()</a></code> function.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples_manually.html">remove_samples_manually</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Label"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.HC_Pool1"</span>, <span class="st">"1.HC_Pool2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 samples removed.</span></span></code></pre></div>
</div>
<div class="section level3" number="5.3">
<h3 id="remove-reference-samples">
<span class="header-section-number">5.3</span> Remove Reference Samples<a class="anchor" aria-label="anchor" href="#remove-reference-samples"></a>
</h3>
<p>And you can remove the reference samples directly using the function <code><a href="../reference/remove_reference_samples.html">remove_reference_samples()</a></code>. But attention: possibly you need them for normalization! That is exactly why we currently keep them!</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_no_refs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_reference_samples.html">remove_reference_samples</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 reference samples removed from the SummarizedExperiment object.</span></span></code></pre></div>
</div>
<div class="section level3" number="5.4">
<h3 id="outlier-detection-via-poma-r-package">
<span class="header-section-number">5.4</span> Outlier Detection via POMA R Package<a class="anchor" aria-label="anchor" href="#outlier-detection-via-poma-r-package"></a>
</h3>
<p>The POMA R package provides a method to detect outliers in proteomics data. The function <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> detects outliers in the data based on the POMA algorithm. The method calculates the euclidean distances (or maximum or manhattan distances) between all sample pairs and then group centroids are computed either as the average distance of group members or the spatial median. The distances of the group members to the corresponding group centroid are then calculated to apply the classical univariate outlier detection formula Q3 + 1.5 * IQR (1.5 is adjustable) to detect group-dependent outliers.</p>
<p>The function returns a list with the following elements: polygon plot, distance boxplot, and the outliers. For further information on the POMA algorithm, please refer to the original publication <span class="citation">(Castellano-Escuder et al. 2021)</span>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poma_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poma_res</span><span class="op">$</span><span class="va">polygon_plot</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-21"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-21-1.png" alt="Polygon plot of the principal coordinates calculated using the POMA algorithm. The original distances are reduced to principal coordinates (see POMA and the vegan::betadisper method)." width="768"><p class="caption">
Figure 5.3: Polygon plot of the principal coordinates calculated using the POMA algorithm. The original distances are reduced to principal coordinates (see POMA and the vegan::betadisper method).
</p>
</div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poma_res</span><span class="op">$</span><span class="va">distance_boxplot</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-22"></span>
<img src="Preprocessing_files/figure-html/unnamed-chunk-22-1.png" alt="Boxplot of the distances of the samples to the group centroid calculated using POMA." width="768"><p class="caption">
Figure 5.4: Boxplot of the distances of the samples to the group centroid calculated using POMA.
</p>
</div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">poma_res</span><span class="op">$</span><span class="va">outliers</span>, </span>
<span>             caption <span class="op">=</span> <span class="st">"Outliers detected by the POMA algorithm."</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>
<span id="tab:unnamed-chunk-23">Table 5.1: </span>Outliers detected by the POMA algorithm.</caption>
<colgroup>
<col width="52%">
<col width="7%">
<col width="23%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">groups</th>
<th align="right">distance_to_centroid</th>
<th align="right">limit_distance</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">Reporter.intensity.corrected.10.Pat_tbc_pool_2</td>
<td align="left">PTB</td>
<td align="right">11.05</td>
<td align="right">10.52</td>
</tr></tbody>
</table>
<p>To remove the outliers detected via the POMA algorithm, just put the data.table of the <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> function into the <code><a href="../reference/remove_POMA_outliers.html">remove_POMA_outliers()</a></code> function.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_POMA_outliers.html">remove_POMA_outliers</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">poma_res</span><span class="op">$</span><span class="va">outliers</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 outlier samples removed.</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="6">
<h2 id="session-info">
<span class="header-section-number">6</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] PRONE_0.99.18               SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.64.0              GenomicRanges_1.56.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.40.1         IRanges_2.38.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.42.1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;   [3] jsonlite_1.8.8              shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;   [5] MultiAssayExperiment_1.30.3 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [7] magick_2.8.4                farver_2.1.2               </span></span>
<span><span class="co">#&gt;   [9] MALDIquant_1.22.2           rmarkdown_2.27             </span></span>
<span><span class="co">#&gt;  [11] GlobalOptions_0.1.2         zlibbioc_1.50.0            </span></span>
<span><span class="co">#&gt;  [13] vctrs_0.6.5                 janitor_2.2.0              </span></span>
<span><span class="co">#&gt;  [15] htmltools_0.5.8.1           S4Arrays_1.4.1             </span></span>
<span><span class="co">#&gt;  [17] SparseArray_1.4.8           mzID_1.42.0                </span></span>
<span><span class="co">#&gt;  [19] sass_0.4.9                  bslib_0.8.0                </span></span>
<span><span class="co">#&gt;  [21] htmlwidgets_1.6.4           plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [23] impute_1.78.0               lubridate_1.9.3            </span></span>
<span><span class="co">#&gt;  [25] cachem_1.1.0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt;  [27] lifecycle_1.0.4             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [29] pkgconfig_2.0.3             Matrix_1.7-0               </span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1                    fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [33] GenomeInfoDbData_1.2.12     snakecase_0.11.1           </span></span>
<span><span class="co">#&gt;  [35] clue_0.3-65                 digest_0.6.36              </span></span>
<span><span class="co">#&gt;  [37] pcaMethods_1.96.0           colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [39] vegan_2.6-6.1               labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [41] fansi_1.0.6                 timechange_0.3.0           </span></span>
<span><span class="co">#&gt;  [43] httr_1.4.7                  abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [45] mgcv_1.9-1                  compiler_4.4.1             </span></span>
<span><span class="co">#&gt;  [47] withr_3.0.1                 doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [49] BiocParallel_1.38.0         UpSetR_1.4.0               </span></span>
<span><span class="co">#&gt;  [51] highr_0.11                  MASS_7.3-61                </span></span>
<span><span class="co">#&gt;  [53] DelayedArray_0.30.1         rjson_0.2.21               </span></span>
<span><span class="co">#&gt;  [55] permute_0.9-7               mzR_2.38.0                 </span></span>
<span><span class="co">#&gt;  [57] tools_4.4.1                 PSMatch_1.8.0              </span></span>
<span><span class="co">#&gt;  [59] glue_1.7.0                  nlme_3.1-165               </span></span>
<span><span class="co">#&gt;  [61] QFeatures_1.14.2            gridtext_0.1.5             </span></span>
<span><span class="co">#&gt;  [63] grid_4.4.1                  cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [65] reshape2_1.4.4              generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [67] gtable_0.3.5                preprocessCore_1.66.0      </span></span>
<span><span class="co">#&gt;  [69] tidyr_1.3.1                 data.table_1.15.4          </span></span>
<span><span class="co">#&gt;  [71] xml2_1.3.6                  utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [73] XVector_0.44.0              foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [75] pillar_1.9.0                stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [77] limma_3.60.4                circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [79] splines_4.4.1               dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [81] ggtext_0.1.2                lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [83] renv_1.0.7                  tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [85] ComplexHeatmap_2.20.0       knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [87] gridExtra_2.3               bookdown_0.40              </span></span>
<span><span class="co">#&gt;  [89] ProtGenerics_1.36.0         xfun_0.46                  </span></span>
<span><span class="co">#&gt;  [91] statmod_1.5.0               MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt;  [93] stringi_1.8.4               UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt;  [95] lazyeval_0.2.2              yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [97] evaluate_0.24.0             codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [99] MsCoreUtils_1.16.1          tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [101] BiocManager_1.30.23         cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [103] affyio_1.74.0               munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [105] jquerylib_0.1.4             Rcpp_1.0.13                </span></span>
<span><span class="co">#&gt; [107] png_0.1-8                   XML_3.99-0.17              </span></span>
<span><span class="co">#&gt; [109] parallel_4.4.1              ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt; [111] dendsort_0.3.4              AnnotationFilter_1.28.0    </span></span>
<span><span class="co">#&gt; [113] scales_1.3.0                affy_1.82.0                </span></span>
<span><span class="co">#&gt; [115] ncdf4_1.22                  purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [117] crayon_1.5.3                POMA_1.14.0                </span></span>
<span><span class="co">#&gt; [119] GetoptLong_1.0.5            rlang_1.1.4                </span></span>
<span><span class="co">#&gt; [121] vsn_3.72.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes in Infected Patients Depending on Tuberculosis Disease State.”</span> <em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
<div id="ref-castellano-escuder_pomashiny:_2021" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domínguez, Francesc Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <span>“<span>POMAShiny</span>: <span>A</span> User-Friendly Web-Based Workflow for Metabolomics and Proteomics Data Analysis.”</span> Edited by Manja Marz. <em>PLOS Computational Biology</em> 17 (7): e1009148. <a href="https://doi.org/10.1371/journal.pcbi.1009148" class="external-link">https://doi.org/10.1371/journal.pcbi.1009148</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
