<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with PRONE • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with PRONE">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/PRONE.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/Normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/Imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Spike_In_Data.html">PRONE with Spike-In Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/daisybio/PRONE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with PRONE</h1>
                        <h4 data-toc-skip class="author">Lis Arend</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/daisybio/PRONE/blob/HEAD/vignettes/PRONE.Rmd" class="external-link"><code>vignettes/PRONE.Rmd</code></a></small>
      <div class="d-none name"><code>PRONE.Rmd</code></div>
    </div>

    
    
<div class="section level2" number="1">
<h2 id="introduction">
<span class="header-section-number">1</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>High-throughput omics data are often affected by systematic biases
introduced throughout all the steps of a clinical study, from sample
collection to quantification. Failure to account for these biases can
lead to erroneous results and misleading conclusions in downstream
analysis. Normalization methods aim to adjust for these biases to make
the actual biological signal more prominent. However, selecting an
appropriate normalization method is challenging due to the wide range of
available approaches. Therefore, a comparative evaluation of
unnormalized and normalized data is essential in identifying an
appropriate normalization strategy for a specific data set. This R
package provides different functions for preprocessing, normalizing, and
evaluating different normalization approaches. Furthermore,
normalization methods can be evaluated on downstream steps, such as
differential expression analysis and statistical enrichment analysis.
Spike-in data sets with known ground truth and real-world data sets of
biological experiments acquired by either tandem mass tag (TMT) or
label-free quantification (LFQ) can be analyzed.</p>
</div>
<div class="section level2" number="2">
<h2 id="installation">
<span class="header-section-number">2</span> Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Official BioC installation instructions</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"PRONE"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load and attach PRONE</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://daisybio.github.io/PRONE/">PRONE</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="workflow">
<span class="header-section-number">3</span> Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p>A six-step workflow was developed in R version 4.2.2 to evaluate the
effectiveness of the previously defined normalization methods on
proteomics data. The workflow incorporates a set of novel functions and
also integrates various methods adopted by state-of-the-art tools.</p>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-3"></span>
<img src="../reference/figures/Workflow_PRONE.png" alt="Six-step workflow of PRONE." width="700px"><p class="caption">
Figure 3.1: Six-step workflow of PRONE.
</p>
</div>
<p>Following the upload of the proteomics data into a SummarizedExperiment
object, proteins with too many missing values can be removed, outlier
samples identified, and normalization carried out. Furthermore, an
exploratory analysis of the performance of normalization methods can be
conducted. Finally, differential expression analysis can be executed to
further evaluate the effectiveness of normalization methods. For data
sets with known ground truth, such as spike-in and simulated data sets,
performance metrics, such as true positives (TPs), false positives
(FPs), and area under the curve (AUC) values, can be computed. The
evaluation of DE results of real-world experiments is based on visual
quality inspection, for instance, using volcano plots, and an
intersection analysis of the DE proteins of different normalization
methods is available.</p>
</div>
<div class="section level2" number="4">
<h2 id="usage">
<span class="header-section-number">4</span> Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>This guide serves as an introduction to the PRONE R package, designed to facilitate the preparation of your data set for utilization of the PRONE package’s functionalities. It begins by delineating the underlying data structure essential for the application of the package, followed by a brief description of how to apply different normalization techniques to your data. Additionally, this tutorial shows how to export the normalized data at the end.</p>
<p>Beyond the scope of this introductory tutorial, PRONE encompassess a broad spectrum of functionalities, ranging from preprocessing steps, imputation, normalization and evaluation of the performance of different normalization techniques, to the identification of differentially expressed proteins. These functionalities are detailed in dedicated vignettes, offering detailed insights and instructions for leveraging full capabilities of the PRONE package:</p>
<ul>
<li><a href="Preprocessing.html">Preprocessing Tutorial</a></li>
<li><a href="Imputation.html">Imputation Tutorial</a></li>
<li><a href="Normalization.html">Normalization Tutorial</a></li>
<li><a href="Differential_Expression.html">Differential Expression Tutorial</a></li>
</ul>
<p>Furthermore, PRONE provides additional functionalities for the analysis of spike-in data sets, which are detailed in the following vignette:</p>
<ul>
<li><a href="Spike_In_Data.html">Spike-In Tutorial</a></li>
</ul>
</div>
<div class="section level2" number="5">
<h2 id="load-data">
<span class="header-section-number">5</span> Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>PRONE uses the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a> class as storage for protein intensities and meta data information on the proteomics data set. Hence, before being able to execute the functionalities of PRONE, the data needs to be saved accordingly. For this, the <code><a href="../reference/load_data.html">load_data()</a></code> function was implemented and requires different parameters which are explained in the following:</p>
<ul>
<li>
<strong>data</strong>: refers to the data.frame containing the protein intensities</li>
<li>
<strong>md</strong>: refers to the data.frame containing the meta-data information</li>
<li>
<strong>protein_column</strong>: refers to the column in the data frame that contains the protein IDs</li>
<li>
<strong>gene_column</strong> (optional): refers to the column in the data frame that contains the gene IDs</li>
<li>
<strong>condition_column</strong> (optional): refers to the column in the meta-data table that contains the condition information - this can also be specified later</li>
<li>
<strong>label_column</strong> (optional): refers to the column in the meta-data table that contains sample label information - sometimes the sample names are very long and this column gives the option to label the samples with shorter names for visualization purposes</li>
</ul>
<p>If you have a tandem mass tag (TMT) data set with samples being measured in different batches than you have to specify the batch information. If reference samples were included in each batch, then additionally specify the samples names of the reference samples.</p>
<ul>
<li>
<strong>batch_column</strong> (optional): refers to the column in the meta-data table that contains the batch information</li>
<li>
<strong>ref_samples</strong> (optional): refers to the samples that should be used as reference samples for normalization</li>
</ul>
<p><b>Attention:</b> You need to make sure that the sample names are saved in a column named “Column” in the meta-data table and are named accordingly in the protein intensity table.</p>
<div class="section level3" number="5.1">
<h3 id="example-1-tmt-data-set">
<span class="header-section-number">5.1</span> Example 1: TMT Data Set<a class="anchor" aria-label="anchor" href="#example-1-tmt-data-set"></a>
</h3>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"tuberculosis_protein_intensities.csv"</span><span class="op">)</span></span>
<span><span class="va">md_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"tuberculosis_metadata.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">md_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">md</span><span class="op">$</span><span class="va">Column</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">md</span><span class="op">$</span><span class="va">Column</span>, <span class="st">" "</span>, <span class="st">"."</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ref_samples</span> <span class="op">&lt;-</span> <span class="va">md</span><span class="op">[</span><span class="va">md</span><span class="op">$</span><span class="va">Group</span> <span class="op">==</span> <span class="st">"ref"</span>,<span class="op">]</span><span class="op">$</span><span class="va">Column</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">md</span>, protein_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, </span>
<span>                gene_column <span class="op">=</span> <span class="st">"Gene.names"</span>, ref_samples <span class="op">=</span> <span class="va">ref_samples</span>, </span>
<span>                batch_column <span class="op">=</span> <span class="st">"Pool"</span>, condition_column <span class="op">=</span> <span class="st">"Group"</span>, </span>
<span>                label_column <span class="op">=</span> <span class="st">"Label"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="5.2">
<h3 id="example-2-label-free-lfq-data-set">
<span class="header-section-number">5.2</span> Example 2: Label-free (LFQ) Data Set<a class="anchor" aria-label="anchor" href="#example-2-label-free-lfq-data-set"></a>
</h3>
<p>The example data set originates from <span class="citation">(Vehmas et al. 2016)</span>. This data set is used for the subsequent examples in this tutorial.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"mouse_liver_cytochrome_P450_protein_intensities.csv"</span><span class="op">)</span></span>
<span><span class="va">md_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"mouse_liver_cytochrome_P450_metadata.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">data_path</span>, check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">md_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">md</span>, protein_column <span class="op">=</span> <span class="st">"Accession"</span>, </span>
<span>                gene_column <span class="op">=</span> <span class="st">"Gene names"</span>, ref_samples <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                batch_column <span class="op">=</span> <span class="cn">NULL</span>, condition_column <span class="op">=</span> <span class="st">"Condition"</span>, </span>
<span>                label_column <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="6">
<h2 id="data-structure">
<span class="header-section-number">6</span> Data Structure<a class="anchor" aria-label="anchor" href="#data-structure"></a>
</h2>
<p>The SummarizedExperiment object contains the protein intensities as “assay”, the meta-data table as “colData”, and additional columns for instance resulting from MaxQuant as “rowData”. Furthermore, information on the different columns, for instance, which columns contains the batch information, can be found in the “metadata” slot.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 1499 12 </span></span>
<span><span class="co">#&gt; metadata(4): condition batch refs label</span></span>
<span><span class="co">#&gt; assays(2): raw log2</span></span>
<span><span class="co">#&gt; rownames(1499): 1 2 ... 1498 1499</span></span>
<span><span class="co">#&gt; rowData names(4): Gene.Names Protein.IDs Peptides used for quantitation</span></span>
<span><span class="co">#&gt;   IDs</span></span>
<span><span class="co">#&gt; colnames(12): 2206_WT 2208_WT ... 2285_Arom 2253_Arom</span></span>
<span><span class="co">#&gt; colData names(3): Column Animal Condition</span></span></code></pre></div>
<p>The different data types can be accessed by using the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays()</a></code> function. Currently, only the raw data and log2-transformed data are stored in the SummarizedExperiment object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 2</span></span>
<span><span class="co">#&gt; names(2): raw log2</span></span></code></pre></div>
</div>
<div class="section level2" number="7">
<h2 id="preprocessing-imputation-normalization-evaluation-and-differential-expression">
<span class="header-section-number">7</span> Preprocessing, Imputation, Normalization, Evaluation, and Differential Expression<a class="anchor" aria-label="anchor" href="#preprocessing-imputation-normalization-evaluation-and-differential-expression"></a>
</h2>
<p>As already mentioned in the introduction section, many functionalities are available in PRONE. All these functionalities are mainly based on the SummarizedExperiment object.</p>
<p>In this tutorial, we will only perform simple normalization of the data using median and LoessF normalization.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"LoessF"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; LoessF completed.</span></span></code></pre></div>
<p>The normalized intensities will be saved as additional assays in the SummarizedExperiment object.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 4</span></span>
<span><span class="co">#&gt; names(4): raw log2 Median LoessF</span></span></code></pre></div>
<p>Again, more information on the individual processes can be find in dedicated vignettes.</p>
</div>
<div class="section level2" number="8">
<h2 id="download-data">
<span class="header-section-number">8</span> Download Data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<p>Finally, you can easily download the normalized data by using the <code><a href="../reference/export_data.html">export_data()</a></code> function. This function will save the specified assays as CSV files and the SummarizedExperiment object as an RDS file in a specified output directory. Make sure that the output directory exists.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.exists</a></span><span class="op">(</span><span class="st">"output/"</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="st">"output/"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/export_data.html">export_data</a></span><span class="op">(</span><span class="va">se</span>, out_dir <span class="op">=</span> <span class="st">"output/"</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"log2"</span>, <span class="st">"Median"</span>, <span class="st">"LoessF"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="9">
<h2 id="session-info">
<span class="header-section-number">9</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] PRONE_1.0.4                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.64.0              GenomicRanges_1.56.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.40.1         IRanges_2.38.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.42.1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [3] clue_0.3-65                 compiler_4.4.1             </span></span>
<span><span class="co">#&gt;  [5] vctrs_0.6.5                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [7] stringr_1.5.1               ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt;  [9] pkgconfig_2.0.3             crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [11] fastmap_1.2.0               XVector_0.44.0             </span></span>
<span><span class="co">#&gt; [13] utf8_1.2.4                  rmarkdown_2.27             </span></span>
<span><span class="co">#&gt; [15] UCSC.utils_1.0.0            preprocessCore_1.66.0      </span></span>
<span><span class="co">#&gt; [17] purrr_1.0.2                 xfun_0.46                  </span></span>
<span><span class="co">#&gt; [19] MultiAssayExperiment_1.30.3 zlibbioc_1.50.0            </span></span>
<span><span class="co">#&gt; [21] cachem_1.1.0                jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt; [23] highr_0.11                  DelayedArray_0.30.1        </span></span>
<span><span class="co">#&gt; [25] BiocParallel_1.38.0         parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [27] cluster_2.1.8.1             R6_2.5.1                   </span></span>
<span><span class="co">#&gt; [29] bslib_0.8.0                 stringi_1.8.4              </span></span>
<span><span class="co">#&gt; [31] limma_3.60.4                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [33] Rcpp_1.0.13                 bookdown_0.40              </span></span>
<span><span class="co">#&gt; [35] iterators_1.0.14            knitr_1.48                 </span></span>
<span><span class="co">#&gt; [37] Matrix_1.7-0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt; [39] tidyselect_1.2.1            rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt; [41] abind_1.4-5                 yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [43] doParallel_1.0.17           ggtext_0.1.2               </span></span>
<span><span class="co">#&gt; [45] codetools_0.2-20            affy_1.82.0                </span></span>
<span><span class="co">#&gt; [47] lattice_0.22-6              tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [49] plyr_1.8.9                  withr_3.0.1                </span></span>
<span><span class="co">#&gt; [51] evaluate_0.24.0             xml2_1.3.6                 </span></span>
<span><span class="co">#&gt; [53] pillar_1.9.0                affyio_1.74.0              </span></span>
<span><span class="co">#&gt; [55] BiocManager_1.30.23         renv_1.0.7                 </span></span>
<span><span class="co">#&gt; [57] foreach_1.5.2               MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt; [59] MALDIquant_1.22.2           ncdf4_1.22                 </span></span>
<span><span class="co">#&gt; [61] generics_0.1.3              ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt; [63] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">#&gt; [65] glue_1.7.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [67] tools_4.4.1                 data.table_1.15.4          </span></span>
<span><span class="co">#&gt; [69] mzID_1.42.0                 QFeatures_1.14.2           </span></span>
<span><span class="co">#&gt; [71] vsn_3.72.0                  mzR_2.38.0                 </span></span>
<span><span class="co">#&gt; [73] XML_3.99-0.17               grid_4.4.1                 </span></span>
<span><span class="co">#&gt; [75] impute_1.78.0               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [77] MsCoreUtils_1.16.1          colorspace_2.1-1           </span></span>
<span><span class="co">#&gt; [79] GenomeInfoDbData_1.2.12     PSMatch_1.8.0              </span></span>
<span><span class="co">#&gt; [81] cli_3.6.3                   fansi_1.0.6                </span></span>
<span><span class="co">#&gt; [83] S4Arrays_1.4.1              dplyr_1.1.4                </span></span>
<span><span class="co">#&gt; [85] AnnotationFilter_1.28.0     pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt; [87] gtable_0.3.5                sass_0.4.9                 </span></span>
<span><span class="co">#&gt; [89] digest_0.6.36               SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt; [91] htmlwidgets_1.6.4           htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [93] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [95] statmod_1.5.0               gridtext_0.1.5             </span></span>
<span><span class="co">#&gt; [97] MASS_7.3-61</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes in Infected Patients Depending on Tuberculosis Disease State.”</span> <em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
<div id="ref-vehmas_liver_2016" class="csl-entry">
Vehmas, Anni P., Marion Adam, Teemu D. Laajala, Gabi Kastenmüller, Cornelia Prehn, Jan Rozman, Claes Ohlsson, et al. 2016. <span>“Liver Lipid Metabolism Is Altered by Increased Circulating Estrogen to Androgen Ratio in Male Mouse.”</span> <em>Journal of Proteomics</em> 133 (February): 66–75. <a href="https://doi.org/10.1016/j.jprot.2015.12.009" class="external-link">https://doi.org/10.1016/j.jprot.2015.12.009</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
